tpool/tpool.h

```diff
@@ -244,7 +260,12 @@ class thread_pool
     assert(m_aio);
     if (use_native_aio)
     {
-      auto new_aio = create_native_aio(max_io);
+      const aio_implementation impl=
+#ifdef LINUX_NATIVE_AIO
```

Comment:
Can this ifdef be put  inside create_native_aio? So that LINUX_NATIVE_AIO would not have to be defined by the API caller, and can be an internal detail of the tpool.

Reply:
I can’t see how it could be moved inside `create_native_aio()`. Here the intention is to replace `m_aio` with an object where some attributes differ, but the type of the implementation is preserved. The function `create_native_aio()` may also be invoked for constructing the very first `thread_pool::m_aio`. This is the cleanest solution that I was able to come up with. Can you post a tested patch that corresponds to what you are suggesting?

Reply:
Ok, will write and post, but this gonna need some time (since I'd need to spend time in an environment I'm not most productive in)